contracts/SpokePool.sol

```diff
@@ -1112,8 +1132,10 @@ abstract contract SpokePool is
         // slither-disable-next-line timestamp
         if (getCurrentTime() - quoteTimestamp > depositQuoteTimeBuffer) revert InvalidQuoteTimestamp();
 
-        // Increment count of deposits so that deposit ID for this spoke pool is unique.
-        uint32 newDepositId = numberOfDeposits++;
+        // Add the last known deposit count to the depositor's ID to ensure that all deposit IDs are unique including
+        // all past deposits which used numberOfDeposits as a counter.
+        uint32 newDepositId = depositCount(depositor);
```

Comment:
something to think about: we don't want other people to be able to spoof the depositor, so maybe do a hash of msg.sender and depositor, to allow this to still be useful for aggregators/smart contracts that deposit on behalf of users while preventing front-running/spoofing to grief fillers